image.write fails unless image.read is called first.

This program fails
```aya
{,
	2 :width;
	2 :height;
	[0 0 0 128 0 0 0 255 0 0 0 255] :data;
	"$(:{sys.wd})/test.bmp" :filename;
} :{image.write}
```
with this error:
```
An unhandled exception occurred. If this is a bug, please submit an issue to https://github.com/nick-paul/aya-lang/issues with the stacktrace below.
=== [ Stacktrace ] ===
java.lang.RuntimeException: java.lang.NullPointerException: Cannot invoke "aya.obj.symbol.Symbol.name()" because "key" is null
	at aya.obj.block.Block.eval(Block.java:169)
	at aya.Aya.run(Aya.java:323)
	at aya.Aya.run(Aya.java:258)
	at aya.Aya.run(Aya.java:103)
Caused by: java.lang.NullPointerException: Cannot invoke "aya.obj.symbol.Symbol.name()" because "key" is null
	at aya.exceptions.runtime.IndexError.<init>(IndexError.java:44)
	at aya.obj.dict.Dict.get(Dict.java:77)
	at aya.util.DictReader.getNumberListEx(DictReader.java:111)
	at aya.ext.image.AyaImage.fromDict(AyaImage.java:44)
	at aya.ext.image.WriteImageInstruction.execute(WriteImageInstruction.java:43)
	at aya.obj.block.Block.eval(Block.java:163)
	... 3 more
```

The main problem lies here:
https://github.com/aya-lang/aya/blob/29629d4c2d8538c5913166d239ef10b8be3a30c0/src/aya/ext/image/AyaImage.java#L44

These static fields are used before the constructor call initializes them
```java
	private static Symbol DATA;
	private static Symbol WIDTH;
	private static Symbol HEIGHT;
```

--- a/src/aya/ext/graphics/CanvasCursorListener.java
+++ b/src/aya/ext/graphics/CanvasCursorListener.java
@@ -17,10 +17,11 @@ import java.util.Set;
 
 public class CanvasCursorListener implements MouseListener, MouseMotionListener {
 
-	private static Symbol X;
-	private static Symbol Y;
-	private static Symbol BUTTON;
-	private static Symbol CLICKS;
+	private static final SymbolTable symbols = Aya.getInstance().getSymbols();
+	private static final Symbol X = symbols.getSymbol("x");
+	private static final Symbol Y = symbols.getSymbol("y");
+	private static final Symbol BUTTON = symbols.getSymbol("button");
+	private static final Symbol CLICKS = symbols.getSymbol("clicks");
 
 	private final Set<Integer> pressedButtons = Collections.synchronizedSet(new HashSet<>());
 	private final SizeBoundedQueue<ClickInfo> clickHistory = new SizeBoundedQueue<>(16);
@@ -28,11 +29,6 @@ public class CanvasCursorListener implements MouseListener, MouseMotionListener 
 	private final SizeBoundedQueue<MoveInfo> moveHistory = new SizeBoundedQueue<>(128);
 
 	public CanvasCursorListener() {
-		SymbolTable symbols = Aya.getInstance().getSymbols();
-		X = symbols.getSymbol("x");
-		Y = symbols.getSymbol("y");
-		BUTTON = symbols.getSymbol("button");
-		CLICKS = symbols.getSymbol("clicks");
 	}
 
 	public List<Integer> getPressedButtons() {

--- a/src/aya/ext/graphics/instruction/keyboard/PressedKeysInstruction.java
+++ b/src/aya/ext/graphics/instruction/keyboard/PressedKeysInstruction.java
@@ -16,20 +16,15 @@ import aya.util.AWTKeyDecoder;
 
 public class PressedKeysInstruction extends GraphicsInstruction {
 
-	private static Symbol KEYCODE;
-	private static Symbol KEY_NAME;
-	private static Symbol LOCATION;
-	private static Symbol LOCATION_NAME;
+	private static final SymbolTable symbols = Aya.getInstance().getSymbols();
+	private static final Symbol KEYCODE = symbols.getSymbol("keycode");
+	private static final Symbol KEY_NAME = symbols.getSymbol("key_name");
+	private static final Symbol LOCATION = symbols.getSymbol("location");
+	private static final Symbol LOCATION_NAME = symbols.getSymbol("location_name");
 
 	public PressedKeysInstruction(CanvasTable canvas_table) {
 		super(canvas_table, "pressed_keys", "N");
 		_doc = "canvas_id: returns a list of dictionaries containing the currently held keys {keycode; key_name; location; location_name;}";
-
-		SymbolTable symbols = Aya.getInstance().getSymbols();
-		KEYCODE = symbols.getSymbol("keycode");
-		KEY_NAME = symbols.getSymbol("key_name");
-		LOCATION = symbols.getSymbol("location");
-		LOCATION_NAME = symbols.getSymbol("location_name");
 	}
 
 	@Override

--- a/src/aya/ext/image/AyaImage.java
+++ b/src/aya/ext/image/AyaImage.java
@@ -19,24 +19,20 @@ import aya.util.DictReader;
 
 public class AyaImage {
 	/** Utility class for loading, storing and writing image in Aya */
-	
+
+	private static final SymbolTable symbols = Aya.getInstance().getSymbols();
+	private static final Symbol DATA = symbols.getSymbol("data");
+	private static final Symbol WIDTH = symbols.getSymbol("width");
+	private static final Symbol HEIGHT = symbols.getSymbol("height");
+
 	private NumberList bytes;
 	private int width;
 	private int height;
-	private static Symbol DATA;
-	private static Symbol WIDTH;
-	private static Symbol HEIGHT;
-	
+
 	public AyaImage(NumberList bytes, int width, int height) {
-		SymbolTable syms = Aya.getInstance().getSymbols();
-		DATA = syms.getSymbol("data");
-		WIDTH = syms.getSymbol("width");
-		HEIGHT = syms.getSymbol("height");
-		
 		this.bytes = bytes;
 		this.width = width;
 		this.height = height;
-
 	}
 	
 	public static AyaImage fromDict(DictReader d) {

